### DISCUSSION TITLE
pull_request event doesn't fire for bot PR

### DISCUSSION THREAD

**Discussion by miles-po** [community/community#56341] [Actions] [answered]
### Select Topic Area

Question

### Body

Trying to set up a higher level event-driven workflow where on first push of a branch makes a PR (populated with JIRA ticket info), and the new PR fires off to deploy an ephemeral AWS stack. So two actions:
1. Auto-create the PR on code push
1. Deploy an AWS stack when the PR is created

If I create the PR manually, the workflow file runs, but if the previous on-push workflow creates the PR, it doesn't run. Because I have the workflow run on PR opened and reopened, I can close and then reopen the PR to run the workflow, but I'd like to have it run automatically. I'd also preferably keep the workflows files separate for clarity/maintenance as I see them as separate concerns.

**Comment by miles-po** [comment, marked as answer]
tl;dr: Don't split the actions as a create PR and then deploy. Use the `create` event to do the main work.

I solved it by doing the following:

```yaml
on:
  create:
  pull_request:
    types:
      - reopened
```
The `create` event—which fires when a branch is pushed for the first time—is paired with the `pull_request` event when reopened.
```yaml
jobs:
  deploy-ephemeral-stack:
    name: Deploy Ephemeral Stack
    if: (github.event.action == 'reopened') || (github.event.ref_type == 'branch')
```
From there, the `pull_request` event will have an `action` property set to "reopened". Alternatively the `create` event will have a `ref_type` property set to "branch".

From there in the case of a new branch pushed to GitHub, the `GITHUB_REF` environment variable will be set to `refs/heads/my_branch` where "my_branch" is the name of the branch just pushed, so you have to remove the `refs/heads/` prefix to get a clean branch name.

For a reopened PR, you can use the `GITHUB_HEAD_REF` environment variable to get the branch name. No prefix trimming needed.
